python/instrumentation/openinference-instrumentation-openai/src/openinference/instrumentation/openai/__init__.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from typing import Any, Collection
+
+from openinference.instrumentation.openai._request import (
+    _AsyncRequest,
+    _Request,
+)
+from openinference.instrumentation.openai.package import _instruments
+from openinference.instrumentation.openai.version import __version__
+from opentelemetry import trace as trace_api
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor  # type: ignore
+from wrapt import wrap_function_wrapper
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+_MODULE = "openai"
+
+
+class OpenAIInstrumentor(BaseInstrumentor):  # type: ignore
+    """
+    An instrumentor for openai.OpenAI.request and openai.AsyncOpenAI.request
+    """
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs: Any) -> None:
+        if (include_extra_attributes := kwargs.get("include_extra_attributes")) is None:
+            include_extra_attributes = True
```

Comment:
i'll remove this. the extra attribute thing is more of an internal two-tier system for attributes, so that we don't lose the important ones if the number of attributes become too numerous (and they start to get dropped by the exporter).

Reply:
the default limit for attributes is 128. the concern is that the flattening of indices can bring up the number of attributes quite quickly